src/api/controllers/ContentController.ts

```diff
@@ -31,63 +32,72 @@ export class ContentController {
   @Authorized("admin")
   @Patch("/contacts")
   async updateContacts(
-    @PartialBody() data: GetContactsContentDto
-  ): Promise<GetContactsContentDto> {
-    ContentTransformer.updateOne("contacts", data);
+    @PartialBody() data: GetContentContactsDto
+  ): Promise<GetContentContactsDto> {
+    await ContentTransformer.updateOne("contacts", data);
     return ContentTransformer.fetchOne("contacts");
   }
 
   @Authorized("admin")
   @Patch("/email")
   async updateEmail(
-    @PartialBody() data: GetEmailContentDto
-  ): Promise<GetEmailContentDto> {
-    ContentTransformer.updateOne("email", data);
+    @PartialBody() data: GetContentEmailDto
+  ): Promise<GetContentEmailDto> {
+    await ContentTransformer.updateOne("email", data);
     return ContentTransformer.fetchOne("email");
   }
 
   @Authorized("admin")
   @Patch("/general")
   async updateGeneral(
-    @PartialBody() data: GetGeneralContentDto
-  ): Promise<GetGeneralContentDto> {
-    ContentTransformer.updateOne("general", data);
+    @PartialBody() data: GetContentGeneralDto
+  ): Promise<GetContentGeneralDto> {
+    await ContentTransformer.updateOne("general", data);
     return ContentTransformer.fetchOne("general");
   }
 
   @Authorized("admin")
   @Patch("/join")
   async updateJoin(
-    @PartialBody() data: GetJoinContentDto
-  ): Promise<GetJoinContentDto> {
-    ContentTransformer.updateOne("join", data);
+    @PartialBody() data: GetContentJoinDto
+  ): Promise<GetContentJoinDto> {
+    if (data.taxRate) {
+      const taxRate = await taxRateUpdateOrCreateDefault(
+        {
+          percentage: data.taxRate
+        },
+        data.taxRateStrapiId
+      );
+      data.taxRateStrapiId = taxRate.id;
```

Comment:
What do you think about setting the tax ID option in the `taxRateUppdateOrCreateDefault` method rather than deferring the saving to the content transformer?

I think it would be a bit cleaner as it would mean the Stripe library doesn't need to know about the `ContentTransformer`, and the API never needs to return the ID in any case. This way you could remove `GetJoinContentDto.taxRateStripeId`.

Reply:
Yes, it also bothered me that I added the dependency to `ContentTransformer` there. I can just query the ID at Strapi. My thought was that someone might want the ability to override the ID , but I guess we don't need to support that.